tensorflow/python/lib/io/file_io.py

```diff
@@ -420,6 +438,22 @@ def get_matching_files_v2(pattern):
             compat.as_bytes(single_filename))
     ]
 
+  def handle_single_pattern(single_pattern):
+    partial_files = [
+      compat.as_str_any(matching_filename)
+      for matching_filename in _pywrap_file_io.GetMatchingFiles(
```

Comment:
Some concern about the performance of this implementation. This should be taking a list of patterns

Also: please provide a benchmark for this implementation

Reply:
Each thread does return a list in this implementation and so I used `functools.reduce` to flatten these lists before returning the result. I think the answer [here](https://stackoverflow.com/a/45323085/6072681) is a valuable reference as for benchmark, which is why I chose it.